### PR TITLE
Fix #4865: Show logbook of GK trackables

### LINKED ISSUE #4865: Show logbook of GK trackable
Of course this feature request was mentioned by OC-users looking at the new Geokret functionality:

The logbook of GK trackables should be loaded as well (like for GC trackables).
I don't know whether the API supports that currently, so I just want to file that as a Feature Request here.

### PR DESCRIPTION
**Pull request by kumy**
GKM api has been reworked, GK logbook is now available.

Not all available informations could be mapped into current `Trackable` `LogEntry`.
ex:
* comments on logs
* move traveled distance
* Lastlogid/lastmoveid (useful for #4867)
* Cache visit count

Example of full details https://api.gkm.kumy.org/gk/47496/details

Todo: add some unit tests with "complete" xml

### CODE REVIEW

**Comment by kumy** [comment]
retest this please

File: main/src/cgeo/geocaching/connector/trackable/GeokretyParser.java [hunk 1]
```diff
@@ -324,4 +442,25 @@ protected static String getType(final int type) {
...
+     */
+    private static String getLastSpottedUsername(final List<LogEntry> logsEntries) {
+        for (LogEntry log: logsEntries) {
+            if (log.getType() == LogType.NOTE) {
+                continue;
+            }
+            if (log.getType() == LogType.GRABBED_IT || log.getType() == LogType.VISIT) {
+                return log.author;
+            }
+            break;
```
Comment (Bananeweizen) [inline main/src/cgeo/geocaching/connector/trackable/GeokretyParser.java]: I'm not really sure what exactly you want to calculate here. If you just want to find any grabbed or visited log, then neither the continue nor the break should be part of the loop, but just the if.
If you want to calculate something else, you still should rework it, to not have a continue _and_ a break in the loop. For me, at most one of those is allowed in a loop to make it understandable.

File: main/src/cgeo/geocaching/connector/trackable/GeokretyParser.java [hunk 2]
```diff
@@ -324,4 +442,25 @@ protected static String getType(final int type) {
...
         return null;
     }
+
+    /**
+     * Find spotted username if last move is Grab/Visit. (Skip comments)
+     *
+     * @param logsEntries
+     *          the log entries to analyze
+     * @return
+     *          The spotted username (or unkonwn)
```
Comment (Bananeweizen) [inline main/src/cgeo/geocaching/connector/trackable/GeokretyParser.java]: typo

File: main/src/cgeo/geocaching/connector/trackable/GeokretyParser.java (hunk 1 above)
Comment (kumy) [inline main/src/cgeo/geocaching/connector/trackable/GeokretyParser.java]: Will rework this part tonight

**Comment by Bananeweizen** [comment]
The change looks good to me besides the above mentioned remarks,

File: main/src/cgeo/geocaching/connector/trackable/GeokretyParser.java [hunk 3]
```diff
@@ -324,4 +442,26 @@ protected static String getType(final int type) {
...
+     */
+    static String getLastSpottedUsername(final List<LogEntry> logsEntries) {
+        for (final LogEntry log: logsEntries) {
+            if (log.getType() == LogType.NOTE) {
+                continue;
+            }
+            if (log.getType() != LogType.GRABBED_IT && log.getType() != LogType.VISIT) {
+                return CgeoApplication.getInstance().getString(R.string.user_unknown);
+            }
+            return log.author;
```
Comment (kumy) [inline main/src/cgeo/geocaching/connector/trackable/GeokretyParser.java]: Hum...

**Pull request merged**